### DISCUSSION TITLE
Send file not work c#

### DISCUSSION THREAD

**Discussion by guiboroni** [salman0ansari/whatsapp-api-nodejs#570] [General]
Good evening guys, 

I'm not able to send the message with the image with c#. I made the code following postman's example but it didn't work. My code is the one below, could you tell me if I'm doing something wrong? Thanks

var client = new RestClient("LINK_MY_SERVER/message/image?key=MY_KEY");                                                                                

            client.Timeout = -1;

            var request = new RestRequest(Method.POST);
            request.AddHeader("Content-Type", "application/json; charset=utf-8");            
            request.AddParameter("id", "55" + "MY NUMBER BY PARAMETER");
            request.AddParameter("caption", msg);            
            request.AlwaysMultipartFormData = true;
            request.AddFile("file", @"C:\20140002.png");                        
            
             client.Execute(request);

**Comment by lostsidedead** [comment]
You can use a simple overload of LinkedResource object that directly takes the file path as the parameter. So there is no need to explicitly load image into memory stream. Of course, this example assumes you have access to the image on the disk -

Here is the complete function code which worked for me -
```
public System.Net.Mail.AlternateView GetAlternateView(string MessageText, string LogoPath, bool bSilent)
{

    try
    {
        MessageText += "<br><img title='' alt='' src='cid:SystemEmail_HTMLLogoPath' />";               

        System.Net.Mail.AlternateView view = System.Net.Mail.AlternateView.CreateAlternateViewFromString(MessageText, null, "text/html");
        System.Net.Mail.LinkedResource linked = new System.Net.Mail.LinkedResource(HttpContext.Current.Request.PhysicalApplicationPath + LogoPath);
        linked.ContentId = "SystemEmail_HTMLLogoPath";
        view.LinkedResources.Add(linked);
        return view;
    }
    catch (Exception ex)
    {
        if (bSilent)
            return null;
        else
            throw ex;
    }
}
```

**Comment by VanProppercr** [comment]
It looks like you're trying to send a message with an image using a REST API in C#. From the code you've shared, it seems that you're using the RestSharp library to make the API call.

There are a couple of things you can check to ensure that your code is working correctly:

Verify that the URL you're using is correct and that it supports sending images.
Check that the API key you're using is valid and has the necessary permissions to send messages with images.
Ensure that the file path of the image you're trying to send is correct and that the file exists in that location.
Make sure that the Content-Type header you're setting is correct and that it supports sending images.
Double-check that the parameters you're passing in the request are correct and that they match the API documentation.
If you've verified all these things and you're still having issues, you can try debugging your code by checking the response from the API. You can use the Execute method of the RestSharp library to get the response from the API and check if there are any error messages or status codes that might give you more information about what's going wrong.
